src/collapse/vsprof.rs

```diff
@@ -0,0 +1,188 @@
+                // There are 3 separate cases to handle regarding the depth:
+                // 1. prev_depth + 1 == depth -> a new function is called, we only need to
+                //    store the function name and the number of times it is called from the
+                //    outer function
+                // 2. prev_depth == depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and replace the top node with our node
+                //    call
+                // 3. prev_depth > depth -> the previous function call was a leaf node, so we
+                //    need to save the current stack and than we need to pop the top nodes
+                //    until the top node is our parent
+                match prev_depth.cmp(&depth) {
+                    // Case 1
+                    Ordering::Less => {
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 2
+                    Ordering::Equal => {
+                        self.write_stack(occurences);
+                        self.stack.pop();
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                    // Case 3
+                    Ordering::Greater => {
+                        self.write_stack(occurences);
+
+                        // Pop (prev_depth - depth - 1) nodes of the stack, this leaves our
+                        // parent as the top node.
+                        for _ in 0..=prev_depth - depth {
+                            self.stack.pop();
+                        }
+
+                        self.stack
+                            .push((function_name.to_string(), number_of_calls));
+                    }
+                }
+            }
+            _ => {
+                return invalid_data_error!("Unable to parse function name from line:\n{}", line);
+            }
+        }
+
+        Ok(())
+    }
+
+    // Store the current stack in `occurences`
+    fn write_stack(&self, occurences: &mut Occurrences) {
+        let (functions, samples): (Vec<_>, Vec<_>) = self.stack.iter().cloned().unzip();
+        if let Some(number_of_calls) = samples.last() {
+            if *number_of_calls > 0 {
+                occurences.insert(functions.join(";"), *number_of_calls);
+            }
+        }
+    }
+}
+
+/// Gets the number from the start of the line. This can either be a number <1000, in which case the
+/// line doesn't contain double quotes, or the number can be >1000, in which case the line does
+/// contain double quotes. In both cases `line` may start with a leading comma, which will be
+/// ignored.
+fn get_next_number(line: &str) -> io::Result<(usize, &str)> {
+    // Trim the leading comma, if any
+    let line = line.trim_start_matches(',');
```

Comment:
Up to you which one you prefer. This one will only remove _one_ comma.
```suggestion
    let line = line.strip_prefix(',').unwrap_or(line);
```